### PR TITLE
Vb_Ball

### PR DESCRIPTION
**Pull request by petrie911**
Splitting from my other pull request. En_Vb_Ball matched and documented.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c [hunk 1]
```diff
@@ -27,17 +38,283 @@ const ActorInit En_Vb_Ball_InitVars = {
...
+    if ((this->actor.bgCheckFlags & 1) != 0) {
+        if (this->actor.velocity.y <= 0.0f) {
+            this->xRotVel = Math_Rand_CenteredFloat(16384.0f);
+            this->yRotVel = Math_Rand_CenteredFloat(16384.0f);
+            angle = Math_atan2f(this->actor.posRot.pos.x, this->actor.posRot.pos.z);
+            this->actor.velocity.x = sinf(angle) * 10.0f;
+            this->actor.velocity.z = cosf(angle) * 10.0f;
+            this->actor.velocity.y = this->actor.velocity.y * -0.5f;
+            if ((this->actor.params & 1) != 0) {
+                Audio_PlaySoundGeneral(0x3927, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c:141]: Please use the constant NA_SE_EN_VALVAISA_LAND here

File: src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c [hunk 2]
```diff
@@ -27,17 +38,283 @@ const ActorInit En_Vb_Ball_InitVars = {
...
+                    EnVbBall_SpawnDebris(globalCtx, bossFd->particles, &debrisPos2, &debrisVel2, &debrisAcc2,
+                                         (s16)Math_Rand_ZeroFloat(12.0f) + 0xF);
+                }
+                Actor_Kill(&this->actor);
+            }
+        }
+        if ((this->collider1.base.atFlags & 2) != 0) {
+            player = PLAYER;
+            this->collider1.base.atFlags &= ~2;
+            Audio_PlayActorSound2(&player->actor, 0x83E);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c:288]: 0x83E corresponds to NA_SE_PL_BODY_HIT

File: src/overlays/actors/ovl_Boss_Fd/z_boss_fd.h [hunk 3]
```diff
@@ -6,9 +6,125 @@
...
+    
+
+typedef struct BossFdParticle {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ Vec3f vel;
+    /* 0x18 */ Vec3f accel;
+    /* 0x24 */ u8 type;
+    /* 0x25 */ u8 timer1;
+    /* 0x26 */ Color_RGB8 color;
+   //* 0x29 */ char pad;
```
Comment (ethteck) [inline src/overlays/actors/ovl_Boss_Fd/z_boss_fd.h:19]: This can be omitted

File: src/overlays/actors/ovl_Boss_Fd/z_boss_fd.h [hunk 4]
```diff
@@ -6,9 +6,125 @@
...
+    /* 0x02B4 */ f32 flightWobbleRate;
+    /* 0x02B8 */ char unk_2B8[4];
+    /* 0x02BC */ Vec3f targetPosition;
+    /* 0x02C8 */ Vec3f holePosition;
+    /* 0x02D4 */ u8 holeIndex;
+    /* 0x02D5 */ u8 eyeState;
+    /* 0x02D6 */ u8 collapsePlatform;
+    /* 0x02D7 */ u8 faceExposed;
+    /* 0x02D8 */ u8 handoffSignal;
+   //* 0x02D9 */ char unk_2D9[3];
```
Comment (ethteck) [inline src/overlays/actors/ovl_Boss_Fd/z_boss_fd.h:87]: this too

File: src/overlays/actors/ovl_Boss_Fd/z_boss_fd.h [hunk 5]
```diff
@@ -6,9 +6,125 @@
...
+    /* 0x1344 */ f32 leftManeScale[30];
+    /* 0x13BC */ Vec3f leftManeHead;    
+    /* 0x13C8 */ f32 flattenMane;
+    /* 0x13CC */ f32 jawOpening;
+    /* 0x13D0 */ s16 bodyFallApart[18];
+    /* 0x13F4 */ Vec3f fireBreathSpawnPoint;
+    /* 0x1400 */ s16 introState1;
+    /* 0x1402 */ s16 introState2;
+    /* 0x1404 */ s16 introCameraState;
+   //* 0x1406 */ char unk_1406[2];
```
Comment (ethteck) [inline src/overlays/actors/ovl_Boss_Fd/z_boss_fd.h:111]: and this

File: src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c [hunk 6]
```diff
@@ -27,17 +38,268 @@ const ActorInit En_Vb_Ball_InitVars = {
...
+void EnVbBall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnVbBall* this = THIS;
+    if (this->actor.params < 0xC8) {
+        Collider_DestroyCylinder(globalCtx, &this->collider1);
+    }
+}
+
+void EnVbBall_SpawnDebris(GlobalContext* globalCtx, BossFdParticle* particle, Vec3f* position, Vec3f* velocity,
+                          Vec3f* acceleration, f32 scale) {
+    s16 i1;
```
Comment (ethteck) [inline src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c:84]: I think for loop iterators should just be called `i` unless there are multiple in the same scope that need different vars (which doesn't seem to be the case here)

File: src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c [hunk 7]
```diff
@@ -27,17 +38,268 @@ const ActorInit En_Vb_Ball_InitVars = {
...
+        this->collider1.dim.height = this->actor.scale.y * 5000.0f;
+        this->collider1.dim.yShift = this->actor.scale.y * -2500.0f;
+        this->xRotVel = Math_Rand_CenteredFloat(8192.0f);
+        this->yRotVel = Math_Rand_CenteredFloat(8192.0f);
+        this->shadowSize = this->actor.scale.y * 68.0f;
+    }
+}
+
+void EnVbBall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnVbBall* this = THIS;
```
Comment (ethteck) [inline src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c:76]: New line after declarations

Comment (ethteck) [reply src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c:76]: (relevant below too)

**Comment by fig02** [comment]
merged into https://github.com/zeldaret/oot/pull/352

**Pull request closed without merging**